src/components/organisms/AdminVenueView/components/Spaces/Spaces.tsx

```diff
@@ -0,0 +1,246 @@
+              {showAdvancedSettings && (
+                <BackgroundSelect venueName={venue.name} mapBackground={""} />
+              )}
+            </div>
+
+            <div>
+              <div
+                className="Spaces__venue-rooms"
+                onClick={() => setShowRooms(!showRooms)}
+              >
+                <div>{numberOfRooms} Rooms</div>
+                <FontAwesomeIcon
+                  icon={!showRooms ? faCaretRight : faCaretDown}
+                />
+              </div>
+
+              {showRooms && (
+                <div>
+                  {venue.rooms?.map((room, index) => {
+                    return (
+                      <div
+                        key={`${index}-${room.title}`}
+                        className="Spaces__venue-room"
+                        onClick={() => setSelectedRoom(room)}
+                      >
+                        <div
+                          className="Spaces__venue-room-logo"
+                          style={{ backgroundImage: `url(${room.image_url})` }}
+                        />
+                        <div className="Spaces__venue-room-title">
+                          {room.title}
+                        </div>
+                      </div>
+                    );
+                  })}
+                </div>
+              )}
+            </div>
+
+            <div
+              className="Spaces__venue-rooms"
+              onClick={() => setShowAddRoom(!showAddRoom)}
+            >
+              <div>Add rooms</div>
+              <FontAwesomeIcon
+                icon={!showAddRoom ? faCaretRight : faCaretDown}
+              />
+            </div>
+            {showAddRoom &&
+              venueRooms.map((venueRoom, index) => (
+                <VenueRoomItem
+                  key={`${venueRoom.text}-${index}`}
+                  text={venueRoom.text}
+                  template={venueRoom.template}
+                  icon={venueRoom.icon}
+                />
+              ))}
+            <div className="Spaces__footer">
+              <div
+                className="Spaces__home-button"
+                onClick={() => history.push("/admin-ng/")}
+              >
+                <FontAwesomeIcon icon={faHome} />
+              </div>
+              <div className="Spaces__nav-buttons">
+                <div
+                  className="Spaces__back-button"
+                  onClick={() => history.push("/admin-ng/")}
+                >
+                  Back
+                </div>
+                <div className="Spaces__next-button" onClick={onClickNext}>
+                  Next
+                </div>
+              </div>
+            </div>
+          </>
+        )}
+      </div>
+      <div className="Spaces__map">
+        <MapPreview
+          isEditing={hasSelectedRoom}
+          mapBackground={venue.mapBackgroundImageUrl}
+          setSelectedRoom={setSelectedRoom}
+          rooms={venue.rooms ?? []}
```

Comment:
```suggestion
          rooms={venue.rooms ?? emptyRoomsArray}
```
emptyRoomsArray should be defined at the top

Reply:
Fixed in: https://github.com/sparkletown/sparkle/pull/1583/commits/f8fa0c67c34ac626919737b4bbd112b4bd715553